pre_build/pre-build-job.yml

```diff
@@ -0,0 +1,76 @@
+- default:
+    name: global
+    prebuild_render: "git@github.com:bamachrn/cccp-pre-build-code.git"
+
+- project:
+    name: cccp-pre-build-job
+    jobs:
+        - centos-container-pipeline-service-pre-build-{{ appid }}-{{ jobid }}-{{ desired_tag }}:
+            appid: {{ appid }}
+            jobid: {{ jobid }}
+            desired_tag: {{ desired_tag }}
+            git_url: {{ git_url }}
+            git_branch: {{ git_branch }}
+            prebuild_branch: {{ appid }}-{{ jobid }}-{{ desired_tag }}
+            prebuild_script: {{ prebuild_script }}
+
+- job-template:
+    name: centos-container-pipeline-service-pre-build-{{ appid }}-{{ jobid }}-{{ desired_tag }}
+    description: |
+        Managed by Jenkins Job Builder, do not edit manually!
+    node: container
+    ci_project: container
+    properties:
+        - github:
+            url: "{git_url}"
+    scm:
+        - git:
+            skip-tag: True
+            git-tool: ci-git
+            branches:
+                - "{git_branch}"
+            remotes:
+              - prebuild-source:
+                  url: "{git_url}"
+    builders:
+        - shell: |
+            export CICO_API_KEY=$(cat ~/duffy.key )
+            while true
+            do
+                cico_output=$(cico node get -f value -c ip_address -c comment)
+                if [ $? -eq 0 ]; then
+                    read CICO_hostname CICO_ssid <<< $cico_output
+                    if  [ ! -z "$CICO_hostname" ]; then
+                        # we got hostname from cico
+                        break
+                    fi
+                    echo "'cico node get' succeed, but can't get hostname from output"
+                fi
+                if [ $n -gt 5 ]; then
+                    # give up after 5 tries
+                    echo "giving up on 'cico node get'"
+                    exit 1
+                fi
+                echo "'cico node get' failed, trying again in 60s ($n/5)"
+                n=$[$n+1]
+                sleep 60
+            done
+            sshopts="-t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -l root"
+            ssh_cmd="ssh $sshopts $CICO_hostname"
+            rsync -e "ssh $sshopts" -Ha .ssh/git_push_priv_key $CICO_hostname:.ssh
+            $ssh_cmd yum -y install rsync git
+            rsync -e "ssh $sshopts" -Ha $(pwd)/ $CICO_hostname:payload
+            $ssh_cmd -t "cd payload && git checkout -b {prebuild_branch} && {prebuild_script}"
+            rtn_code=$?
+
+            if [ $rtn_code -eq 0 ]; then
+              $ssh_cmd -t "rm -f .gitignore && "
+              "git remote add prebuild-render {prebuild_render} && "
+              "git add -A && "
+              "git commit -am 'Added changes from pre-build' && "
```

Comment:
if we can collect the commit ID from original source repository and embed it the commit message of rendered git, it would help map the changes. Say in long run, there would be only similar commit messages added. This is not needed right now, but you can add a #TODO here and we can visit this in upcoming iterations.

Reply:
lets do it in next iteration